packages/tldraw/src/lib/shapes/geo/GeoShapeUtil.tsx

```diff
@@ -339,6 +341,39 @@ export class GeoShapeUtil extends BaseBoxShapeUtil<TLGeoShape> {
 		})
 	}
 
+	override getHandleSnapGeometry(shape: TLGeoShape): HandleSnapGeometry {
+		const geometry = this.getGeometry(shape)
+		// we only want to snap handles to the outline of the shape - not to its label etc.
+		const outline = geometry.children[0]
+		switch (shape.props.geo) {
+			case 'arrow-down':
+			case 'arrow-left':
+			case 'arrow-right':
+			case 'arrow-up':
+			case 'check-box':
+			case 'diamond':
+			case 'hexagon':
+			case 'octagon':
+			case 'pentagon':
+			case 'rectangle':
+			case 'rhombus':
+			case 'rhombus-2':
+			case 'star':
+			case 'trapezoid':
+			case 'triangle':
```

Comment:
extremely nonblocking:

triangle might want a special case for its center point since it's bounding box is not center aligned with the triangle itself

![Kapture 2024-02-15 at 11 41 03](https://github.com/tldraw/tldraw/assets/1242537/c7e3c135-6676-4143-983e-7eb2529afab8)